File: src/main/java/no/nav/tjenestepensjon/simulering/soap/SoapClientConfig.java
package no.nav.tjenestepensjon.simulering.soap;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.oxm.jaxb.Jaxb2Marshaller;
import org.springframework.ws.client.core.WebServiceTemplate;

@Configuration
public class SoapClientConfig {

    @Value("${PROVIDER_URI}")
    private String providerUri;

    @Bean
    Jaxb2Marshaller jaxb2Marshaller() {
        Jaxb2Marshaller jaxb2Marshaller = new Jaxb2Marshaller();
        jaxb2Marshaller.setContextPath("no.nav.ekstern.pensjon.tjenester.tjenestepensjonsimulering.v1");
        return jaxb2Marshaller;
    }

    @Bean
    public WebServiceTemplate webServiceTemplate(SoapFaultHandler soapFaultHandler) {
        WebServiceTemplate webServiceTemplate = new WebServiceTemplate();
        webServiceTemplate.setMarshaller(jaxb2Marshaller());
        webServiceTemplate.setUnmarshaller(jaxb2Marshaller());
        webServiceTemplate.setDefaultUri(providerUri);
        webServiceTemplate.setFaultMessageResolver(soapFaultHandler);
        webServiceTemplate.setCheckConnectionForFault(false);
        return webServiceTemplate;
    }
}

File: src/main/kotlin/no/nav/tjenestepensjon/simulering/soap/SoapClientConfig.kt
package no.nav.tjenestepensjon.simulering.soap

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.oxm.jaxb.Jaxb2Marshaller
import org.springframework.ws.client.core.WebServiceTemplate

@Configuration
class SoapClientConfig(
        @Value("\${PROVIDER_URI}") val providerUri: String
) {

    @Bean
    fun jaxb2Marshaller() = Jaxb2Marshaller()
            .apply {
                setPackagesToScan("no.nav.tjenestepensjon.simulering.model.v1")
            }

    @Bean
    fun webServiceTemplate() =
            WebServiceTemplate().apply {
                marshaller = jaxb2Marshaller()
                unmarshaller = jaxb2Marshaller()
                defaultUri = providerUri
                faultMessageResolver = SoapFaultHandler(jaxb2Marshaller())
                setCheckConnectionForFault(false)
                setCheckConnectionForError(false)
            }
}